src/main/java/com/checkmarx/flow/controller/GitHubController.java

```diff
@@ -114,6 +118,21 @@ public String pingRequest(
         } catch (IOException e) {
             throw new MachinaRuntimeException(e);
         }
+
+        if (properties != null) {
```

Comment:
Basically you are right, but it's not a duplication, but a similar code. Each piece deals with different Event object, so it won't be easy to extract both sections to a central place, since they don't have a common point.

Reply:
I moved the logic from the controller into the GitHubService